primap2/_overview.py

```diff
@@ -0,0 +1,98 @@
+"""Summarization and descriptive statistics functions to get an overview of a data
```

Comment:
A nice feature would be to be able to treat the time coordinate differently such that you have the option to have just 1 or 0 from the time coordinate. I  might want to know e.g. how many time series are present for a sector. In this case counting the individual time points is not helpful and selecting an individual time point doesn't give the full picture as there might be nan values for that year despite data being available for other time points.

Reply:
This would be functionality similar to `pr.sum_skip_all_na()` (in spirit), right? So, where `sum_skip_all_na` gives the option to sum along some dimension skipping all timeseries that are completely missing (all NA), you propose for the coverage() function an option to only count those timeseries that have at least one value, i.e. those that would not be skipped by `sum_skip_all_na`, right?

I have the feeling that there might be a more fundamental aggregation function like `sum_skip_all_na`, something like a `count_skip_all_na` or so, which would give this functionality also in other contexts. I'll think a bit about it.

Reply:
So it could already be done by first collapsing the time dimensions using sum_skip_all_na and then running pr.coverage on the result. That's fine with me. It's good not to overload functions if it can already be done. But we have to take care to write a good documentation if users have to run several functions inn a row to achieve certain things (examples maybe).

Reply:
No, it is not yet possible, because the aggregation function we need is `all`, not `sum`. I will experiment with a new `all_skip_all_na` function (urgh, the name is consistent with `sum_skip_all_na`, but not very pretty) and see if it leads to a nice API.